internal/beater/beater.go

```diff
@@ -312,14 +312,23 @@ func (r *reloader) reloadOutput(config *reload.ConfigWithMeta) error {
 		}
 	}
 	r.mu.Lock()
+	defer r.mu.Unlock()
 	r.outputConfig = outputConfig
-	r.mu.Unlock()
 	return r.reload()
 }
 
-func (r *reloader) reload() error {
+func (r *reloader) reloadOnce() error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
+
+	return r.reload()
```

Comment:
Seeing as this is only called in one place, should we inline it there?

Reply:
I think it's also called in `reloadOutput`, on line 317, right?

Reply:
`reload` is called directly there, because `reloadOutput` already holds the mutex

Reply:
Removed `reloadOnce` and inlined the mutex acquisition for standalone mode in `run`.